discovery/syncer.go

```diff
@@ -669,10 +680,70 @@ func (g *GossipSyncer) synchronizeChanIDs() (bool, error) {
 	return false, err
 }
 
+// isLegacyReplyChannelRange determines where a ReplyChannelRange message is
+// considered legacy. There was a point where lnd used to include the same query
+// over multiple replies, rather than including the portion of the query the
+// reply is handling. We'll use this as a way of detecting whether we are
+// communicating with a legacy node so we can properly sync with them.
+func isLegacyReplyChannelRange(query *lnwire.QueryChannelRange,
+	reply *lnwire.ReplyChannelRange) bool {
+
+	return reply.QueryChannelRange == *query
+}
+
+// lastHeightInRange is a helper function to return the last block height of a
+// QueryChannelRange message.
+func lastHeightInRange(msg *lnwire.QueryChannelRange) uint32 {
```

Comment:
could be method on `lnwire.QueryChannelRange`?

Reply:
Good idea, done.